mesonbuild/build.py

```diff
@@ -1120,10 +1050,10 @@ def process_kwargs(self, kwargs):
         self.add_deps(deplist)
         # If an item in this list is False, the output corresponding to
         # the list index of that item will not be installed
-        self.install_dir = typeslistify(kwargs.get('install_dir', []),
+        self.install_dir = typeslistify(kwargs.get('install_dir') or [],
                                         (str, bool))
         self.install_mode = kwargs.get('install_mode', None)
-        self.install_tag = stringlistify(kwargs.get('install_tag', [None]))
+        self.install_tag = [kwargs['install_tag']] if kwargs.get('install_tag') else [None]
```

Comment:
Could use `_process_install_tag(kwargs['install_tag'], 1)` here to be consistent with CustomTarget.

Reply:
ah yes, good catch